streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -231,8 +231,10 @@ public void onCompletion(final RecordMetadata metadata,
                 }
             });
         } catch (final RuntimeException uncaughtException) {
-            if (uncaughtException instanceof KafkaException &&
-                uncaughtException.getCause() instanceof ProducerFencedException) {
+            if (uncaughtException instanceof KafkaException &&(
+                uncaughtException.getCause() instanceof ProducerFencedException ||
+                uncaughtException.getCause() instanceof UnknownProducerIdException
```

Comment:
The error message of `RecoverableClientException` does not match any longer... (I would recommend to _not_ add the error name to begin with, as the root cause exception is pass into the constructor anyway)